### ISSUE TITLE
Fix false positives for mixins in max-empty-lines

### ISSUE THREAD

**Issue by iliyaZelenko** [stylelint/stylelint#4180]
> Clearly describe the bug

Strange error: "Expected no more than 1 empty line (max-empty-lines)"

> Which rule, if any, is the bug related to?

`max-empty-lines`

> What CSS is needed to reproduce the bug?

```sass
div
  color: #fff

@mixin post-grid($columns, $width)
  color: #fff

span
  color: #fff
```

> What stylelint configuration is needed to reproduce the bug?

```yaml
extends:
  - stylelint-config-standard

rules:
  "block-closing-brace-empty-line-before": null
  "block-closing-brace-newline-after": null
  "block-closing-brace-newline-before": null
  "block-closing-brace-space-before": null
  "block-opening-brace-newline-after": null
  "block-opening-brace-space-after": null
  "block-opening-brace-space-before": null
  "declaration-block-semicolon-newline-after": null
  "declaration-block-semicolon-space-after": null
  "declaration-block-semicolon-space-before": null
  "declaration-block-trailing-semicolon": null
```

> Which version of stylelint are you using?

`10.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint ./src/styles/**/*.sass` and error in my IDE.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SASS.

> What did you expect to happen?

No errors.

> What actually happened (e.g. what warnings or errors did you get)?

"Expected no more than 1 empty line (max-empty-lines)"

![image](https://user-images.githubusercontent.com/13103045/62123015-a8b63a80-b2cf-11e9-901e-1ac35b0ba5ad.png)

Note: with this code I have no error (no selector at the end):

```sass
div
  color: #fff

@mixin post-grid($columns, $width)
  color: #fff
```

**Comment by jeddy3** [comment]
Thanks for the report and for using the template. I can replicate this. 

I suspect the problem lies within the interplay of stylelint and the postcss-sass parser. I'll label up the issue as "help wanted" as I don't believe any of the core team uses the sass syntax.

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
